### ISSUE TITLE
Named params as object are not supported

### ISSUE THREAD

**Issue by garcia556** [daywiss/json-rpc-core#2]
Hello,

As per JSON-RPC specification:

> 4.2 Parameter Structures
> If present, parameters for the rpc call MUST be provided as a Structured value. Either by-position through an Array or by-name through an Object.

> by-name: params MUST be an Object, with member names that match the Server expected parameter names. The absence of expected names MAY result in an error being generated. The names MUST match exactly, including case, to the method's expected parameters.

Only by-position Array is currently supported via:

```javascript
   function callMethod(method,params,context){
     if(lodash.isFunction(method)){
       return method.apply(context,params)
     }
     return method
   }
```

as Function.apply accepts only Array, not Object with named keys.

Would be nice to support such Object as well as per the specs.
Something like this could work (copy-paste from jayson project):

```javascript
var rpcFunc = function(a1, b2, c3)
{
  console.log(a1, b2, c3);
}

var getParameterNames = function(func) {
  if(typeof(func) !== 'function') {
    return [];
  }

  var body = func.toString();
  var args = /^(?:function )*.*?\((.+?)\)/.exec(body);
  if(!args) {
    return [];
  }

  var list = (args.pop() || '').split(',');
  return list.map(function(arg) { return arg.trim(); });
};

var params = { a1: 100, b2: "str", c3: true };
var paramsArr = getParameterNames(rpcFunc).map(function(name) { return params[name]; });

console.log(paramsArr);

```

**Comment by daywiss** [comment]
Wow that is pretty crazy, I think i see whats going on though.  So you need to stringify the actual rpc function to parse out its parameter names, then create an array from the object in the correct order.  

If you actually want to send a single object as the first parameter, in that case I guess you can no longer send it by itself, but have it named something, like this?
```
var params = { params: { /* params object */}}
```

**Comment by garcia556** [comment]
Yeah one summary object would work. But I was migrating from *jayson* to some transport-agnostic library so didn't want to rewrite actual handlers because of JSON-RPC module I use.

And JSON-RPC spec looks pretty strict: **If present, parameters for the rpc call MUST be provided as a Structured value. Either by-position through an Array or by-name through an Object.**

I was using json-rpc-core in production for a while however I noticed that CPU usage eventually goes to 100%... I didn't have enough time to investigate this properly or to raise as a separate issue but from the profiling I made function calls related to **highland** library were to blame. 

Maybe I just used it in some wrong way using async/await but eventually I switched back to *jayson* internal interfaces to handle JSON-RPC requests back and forth where named params work fine.

So basically no issues with json-rpc-core for me anymore. I tried to reproduce the CPU issue but in non-prod that appeared to be not very easy.

Thanks for the library though 👍 

As for the code I used when getting 100% CPU it was something like this:
```javascript
const jsonRpc = require("json-rpc-core");

class WsHandler
{
	constructor(clients, ws, api)
	{
		this.clients		= clients;
		this.ws				= ws;
		this.api			= api;
	}

	onConnect()
	{
		let _self = this;
		let ws = _self.ws;

		_self.rpc = jsonRpc(_self.api);

		ws.on("message", (message) => {
			_self.rpc.write(JSON.stringify(message));
		});														// got data from the client

		// got data to send to the client
		_self.rpc.on("data", (data) => {
			ws.send(JSON.stringify(data));
		});
	}
}
```

Nothing special, just sending/receiving data through WebSockets.
